src/service/viewport/viewport-impl.js

```diff
@@ -139,15 +139,12 @@ export class ViewportImpl {
     /** @private {string|undefined} */
     this.originalViewportMetaString_ = undefined;
 
-    /** @private @const {!FixedLayer} */
-    this.fixedLayer_ = new FixedLayer(
-      ampdoc,
-      this.vsync_,
-      this.binding_.getBorderTop(),
-      this.paddingTop_,
-      this.binding_.requiresFixedLayerTransfer()
-    );
-    ampdoc.whenReady().then(() => this.fixedLayer_.setup());
+    /** @private {?FixedLayer} */
+    this.fixedLayer_ = null;
+
+    if (!MOVE_FIXED_LAYER) {
```

Comment:
Move this check to inside the method, so it can be DCE easier.

Reply:
@jridgewell the same method is used by the amp-viewer-integration so I don't think that would work? Its the invocation here that is conditional

Reply:
👍